### PR TITLE
Modifications to get_optics to survive from N/A output power value

### PR DESCRIPTION
**Pull request by kaage**
Part of: https://github.com/napalm-automation/napalm-base/issues/237

### CODE REVIEW

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -592,6 +592,13 @@ def get_optics(self):
             port_detail['physical_channels'] = {}
             port_detail['physical_channels']['channel'] = []
 
+            # If interface is shutdown it returns "N/A" as output power.
+            # Converting that to -40.0 float
+            try:
+                float(output_power)
+            except ValueError:
+                output_power = -40.0
```
Comment (dbarrosop) [inline napalm_ios/ios.py:600]: Out of curiosity, why -40? Is that an insane value that clearly indicates it is not supported/available? I am not an optics person myself :P

In any case, I am fine with having such a value if it makes sense but we should make it clear here:

https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/base.py#L1332

Comment (kaage) [reply napalm_ios/ios.py:600]: Yeah, I was thinking about this. I'm wondering whether it is good to use 0.0 as default value for optical powers as it is (in some cases) completely valid value. So 0.0 doesn't indicate not supported(available) see [Cisco 10GBASE SFP+ Modules datasheet ](http://www.cisco.com/c/en/us/products/collateral/interfaces-modules/transceiver-modules/data_sheet_c78-455693.pdf) table 2 for example. Cisco uses value of -40.0 at Rx side to indicate "no light". Nevermind, as technology evolves somebody will create more sensitive receivers and  -40.0 will be again valid value. Should we default to -100 or something?

Even you are fine with such value I think we should have it consistent across platforms and all values of Rx, Tx, instant, max, min, avg.

Comment (dbarrosop) [reply napalm_ios/ios.py:600]: Yeah, I am fine with -100 as a sort of "infinite" value. Eventually we will move this to the classes in `napalm-yang` and those have support to leave empty values.

Before moving forward I'd like @mirceaulinic's opinion though but if he agrees we can probably create a few issues to (1) document the behavior in `napalm-base` and (2) update the rest of the drivers as well.

**Comment by ktbyers** [comment]
This looks good to me...so can integrate once we decide default power value for "N/A".

**Comment by dbarrosop** [comment]
FYI

I have created an Epic:

https://github.com/napalm-automation/napalm-base#boards?epics=53213638_237&repos=76636028,67531943,32737992,49192168,76226659,53213638,70275626,74009031,79204958,53214857,53658916,54102492,53492012,53583872,64762070,53258608,54017316,58233121,53928100,61367331,56185510,74008814,57410375,69223886,68358654,69226025,67635081

**Comment by ktbyers** [comment]
@kaage Can you update this to make the "infinite" value be -100?

File: napalm_ios/ios.py (hunk 1 above)
Comment (mirceaulinic) [reply napalm_ios/ios.py:600]: Yes, that makes sense to me.

**Comment by ktbyers** [comment]
I will integrate and fix the -100 references.

**Pull request merged**